File: 1주차/PGS_모의고사/seoheun.py
def solution(answers):
    answer = { 1: 0, 2: 0, 3: 0 }
    
    student1 = [1, 2, 3, 4, 5] # 5
    student2 = [2, 1, 2, 3, 2, 4, 2, 5] # 8
    student3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5] # 10
    
    for i in range(len(answers)):
        if answers[i] == student1[i % 5]:
            answer[1] += 1
        if answers[i] == student2[i % 8]:
            answer[2] += 1
        if answers[i] == student3[i % 10]:
            answer[3] += 1
    
    result = answer[1]
    keyList = []
        
    for key in range(1, 4):
        if result < answer[key]:
            result = answer[key]
            keyList = []
            keyList.append(key)
        elif result == answer[key]:
            keyList.append(key)

    keyList.sort()
            
    return keyList

File: 6주차/PGS_단속카메라/seoheun.js
// 1차 시도
function solution(routes) {
    var answer = 0;

    const findAnswer = (routeList) => {
        routeList.sort((a, b) => a[0] - b[0]);
        let startLine = routes[0];
        routeList.shift();

        routeList.sort((a, b) => b[1] - a[1]);
        let endLine = routes[0];
        routeList.shift();

        answer = answer + 2;

        let outOfStart = routes.filter((value) => startLine[1] < value[0]);
        let outOfEnd = routes.filter((value) => value[1] < endLine[0]);
        let new_routes = [...outOfStart, ...outOfEnd];

        if (new_routes.length < 3) {
            return answer;
        } else {
            findAnswer(new_routes);
        }
    };

    let result = findAnswer(routes);

    return result;
}

// AI 피드백을 받은 2차 시도 - 성공
function solution(routes) {
    var answer = 0;

    while (routes.length > 0) {
        routes.sort((a, b) => a[1] - b[1]);
        let startLine = routes[0];
        routes.shift();

        let routesLeft = routes.filter((value) => value[0] > startLine[1]);
        if (routesLeft.length > 0) {
            answer++;
        }

        routes = routesLeft;
    }

    return answer + 1;
}
